Gemfile

```diff
@@ -20,7 +20,7 @@ gem 'pimpmychangelog', '>= 0.1.2'
 gem 'pry'
 if RUBY_PLATFORM != 'java'
   # There's a few incompatibilities between pry/pry-byebug on older Rubies
-  gem 'pry-byebug' if RUBY_VERSION >= '2.6.0' && RUBY_ENGINE != 'truffleruby'
+  gem 'pry-byebug' if RUBY_VERSION >= '2.6.0' && RUBY_VERSION < '3.0.0' && RUBY_ENGINE != 'truffleruby'
```

Comment:
What's the reason behind this change?

Reply:
I had some problems installing byebug on 3.0 for some reason (not in all cases though). This ended up in the PR. That said, it's not the first time I had issues with byebug.

Reply:
@marcotc Here's the issue (representative excerpt, there are multiple points of such failure), it shows up with recent compilers:

```
compiling breakpoint.c
In file included from breakpoint.c:1:
In file included from ./byebug.h:4:
In file included from /nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby.h:38:
In file included from /nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/ruby.h:23:
In file included from /nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/defines.h:73:
In file included from /nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/backward/2/attributes.h:43:
In file included from /nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/internal/attr/pure.h:25:
/nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/assert.h:132:1: error: '__declspec' attributes are not enabled; use '-fdeclspec' or '-fms-extensions' to enable support for
__declspec attributes
RBIMPL_ATTR_NORETURN()
^
/nix/store/d8pmxkyckrs8kpz60bj38cq13hhdmjbc-ruby-3.0.2/include/ruby-3.0.0/ruby/internal/attr/noreturn.h:29:33: note: expanded from macro 'RBIMPL_ATTR_NORETURN'
# define RBIMPL_ATTR_NORETURN() __declspec(noreturn)
```

Reply:
This is pretty interesting/weird... Could you share your `Gemfile.lock` that led to this issue (if any?). I'm curious which versions of pry/pry-byebug were being installed.

I tried to reproduce this on a NixOS docker image and it worked for me.

Reply:
Well, no since it's happening at the `bundle install` stage, so no `Gemfile.lock`. But it's byebug 11.1.3 and pry-byebug 3.8.0.

Reply:
Are you using the NixOS Ruby 3.0.2, or did you build your own?

Reply:
nix's, but on darwin

Reply:
Ohh... Hmmm there doesn't seem to be a great way to try it out without changing my system 😰